quint/src/runtime/compile.ts

```diff
@@ -82,15 +82,17 @@ export function newCompilationState(): CompilationState {
   }
 }
 
-export function errorContextFromMessage(errors: ErrorMessage[]): CompilationContext {
-  return {
-    lookupTable: new Map(),
-    syntaxErrors: errors,
-    analysisErrors: [],
-    compileErrors: [],
-    getRuntimeErrors: () => [],
-    compilationState: newCompilationState(),
-    evaluationState: newEvaluationState(),
+export function errorContextFromMessage(listener: ExecutionListener): (errors: ErrorMessage[]) => CompilationContext {
+  return (errs: ErrorMessage[]) => {
+    return {
+      lookupTable: new Map(),
+      syntaxErrors: errs,
+      analysisErrors: [],
+      compileErrors: [],
+      getRuntimeErrors: () => [],
+      compilationState: newCompilationState(),
+      evaluationState: newEvaluationState(listener),
+    }
```

Comment:
Idk if we want to instantiate a new empty compilation and evaluation state for every error message. I believe we could do something like this:
```suggestion
export function errorContextFromMessage(listener: ExecutionListener): (errors: ErrorMessage[]) => CompilationContext {
  const compilationState = newCompilationState()
  const evaluationState = newEvaluationState(listener)
  return (errs: ErrorMessage[]) => {
    return {
      lookupTable: new Map(),
      syntaxErrors: errs,
      analysisErrors: [],
      compileErrors: [],
      getRuntimeErrors: () => [],
      compilationState,
      evaluationState,
    }
```